src/support/lockedpool.cpp

```diff
@@ -133,6 +133,12 @@ Arena::Stats Arena::stats() const
     return r;
 }
 
+bool Arena::addressInArena(void *ptr_in) const
+{
+    uintptr_t ptr = reinterpret_cast<uintptr_t>(ptr_in);
```

Comment:
Wouldn't it be easier to store a begin and end void*, and then compare (ptr_in >= ptr_base && ptr_in < ptr_end)? That way you don't need the reinterpret cast.